### DISCUSSION TITLE
How impossible would it be to run react-native-Windows apps, on Xbox One?

### DISCUSSION THREAD

**Discussion by Seanmclem** [microsoft/react-native-windows#8651] [Q&A] [answered]
I understand that Xbox One apps, to some extent, are basically just Windows apps. Including games. With react native for Windows directly targeting Windows application API- how crazy would it be to run RNW apps on an Xbox One? I assume it wouldn't "just work", and may require some changes and blessings on the Xbox side, and some updates on the RNW side, but how far off would it be?

**Comment by chrisglein** [comment, marked as answer]
Shouldn't be impossible at all. It should just work as long as you select Xbox as one of your outputs when you put together your app packages. I haven't done it myself, but here are the UWP docs on this: https://docs.microsoft.com/en-us/windows/uwp/xbox-apps/
Of note, the Store on Xbox is written in RNW, so it definitely should work ;)

**Reply by Seanmclem** [reply]
Isnt that the "Xbox App", on Windows? Not the Store app on the actual Xbox.

**Reply by Seanmclem** [reply]
It seems like the UWP project is configured by a react native CLI? I wonder if I can select the Xbox Option.

Also, I saw someone mentioned once that UWP apps on Xbox are restricted to 25% processing power? Is that right? Or is that only for the webview?

**Reply by Seanmclem** [reply]
@chrisglein

**Reply by chrisglein** [reply]
There are multiple Xbox apps, some on PC, some on console. Multiple are built with NR, but yes specifically the store app on the Xbox itself is RN.

If you build a UWP app the Universal part means it can deploy to Xbox. There should be an option when assembling your app package when going through the store prep via Visual Studio.

I'm not familiar with what the performance restrictions are for Xbox apps.

**Reply by Seanmclem** [reply]
Seems like this article describes the limitations, under software limitations. Seems like apps are more limited than games, capped at 1 GB or 5 GB of RAM respectively. With other GPU/CPU limitations under Hardware. 
https://www.techrepublic.com/article/xbox-series-s-and-x-developer-mode-3-things-you-can-do-with-it-and-3-you-cant/

Which is fine. I'd probably be building a game with it anyway.

**Comment by gaetandezeiraud** [comment]
Any idea how to make the app (debug) running on Xbox connect to metro (on PC)?

**Reply by Seanmclem** [reply]
The same difficulty can happen on phones, mostly Android. So maybe Google for that and solve the same way? The metro server by default runs on localhost, but the xbox does not. 

Seems like most methods require the dev menu? Any way to open that on Xbox?
https://stackoverflow.com/a/64080669/3705183

Or this one, maybe we can programmatically set the IP in a file before running. 
https://github.com/facebook/react-native/issues/10715 

Please post here if you figure it out.

**Comment by acoates-ms** [comment]
https://microsoft.github.io/react-native-windows/docs/ReactInstanceSettings#sourcebundlehost Will allow you to specify another computer for the xbox to connect to metro from.    `Shift+Ctrl+D` should bring up the dev menu, the same as on windows.
